### ISSUE TITLE
Build more robust way of setting times and timezone for webinars

### ISSUE THREAD

**Issue by cjyabraham** [cncf/cncf.io#162]
As discussed, have proper dropdowns for setting the time and timezones of webinars.  

Note: for LFPH, this will need to be merged by hand into the MU plugin which is now forked.

**Comment by cjyabraham** [comment]
What about just using google for tz conversion: [https://www.google.com/search?q=12pm+kst+is+what+here](https://www.google.com/search?q=12pm+kst+is+what+here)

That way we don’t need to map timezones to some weird numbering that other site uses.

**Comment by cjyabraham** [comment]
I've rewritten this to take a time and period from a dropdown.  Timezone is still free text:
<img width="219" alt="Screen Shot 2020-06-11 at 4 33 27 PM" src="https://user-images.githubusercontent.com/463160/84363455-576e1b80-ac01-11ea-98ea-fde096c35d08.png">

There are a few open questions I'm still working on:
1. I did the "Convert to your timezone" links [using google](https://www.google.com/search?q=7:00+PM+PST+is+what+here), however, it doesn't take a date so may produce misleading results on the odd occasion when the actual date is in a different daylight savings time.  Maybe we can live with this for now.
2. I could potentially have timezone be chosen from a list but haven't found a good list yet to use that works for PHP.

I've made a note that when we migrate, upcoming webinar times will need to be manually entered here.

LMK what you think @thetwopct .

**Comment by cjyabraham** [comment]
I've added the full list dropdown to choose timezone from.  It's a little unwieldy but arguably better than the free-text entry.  I think this issue is more or less complete.  @thetwopct would you test and confirm?

**Comment by thetwopct** [comment]
Looks good. The timezones list is extremely comprehensive!

I don't consider the Google conversion useful at all. It's like telling someone to Google the answer - somewhat accurate and offensive at the same time. So would prefer to stick with the method I had implemented. You said my method didn't work sometimes, but I couldn't reproduce, can you let me know the issues you experienced with it and I can just fix that?

**Comment by cjyabraham** [comment]
The issue with the service you used was that each timezone had an index that was unique to that service.  We'd need to lookup each one and store the correct index for each possible timezone in order to pass in the right one with a lookup.  

Why do you think the Google one is not useful?  It translates the time into the person's local timezone.  I get that it might be offputting to some people, however, so I could look around for an alternative service that might solve all of the above issues...

**Comment by thetwopct** [comment]
The Google conversion requires the user to think too much about days as it only converts the time. Google adds the day to the result, based on the users current day.

i.e. 
This [post](https://dev-cncfci.pantheonsite.io/webinars/the-definitive-checklist-for-delivering-reliable-kubernetes-based-applications-2/), the webinar is on a Monday, the link to Google is: https://www.google.com/search?q=10:00+AM+PDT+is+what+here

Google tells me "18:00 Thursday, in London, UK" so without being careful I would assume that the webinar is at 6pm Thursday. This method also doesn't pay attention to summer time/dateline changes, which only happen a few times a year they do happen over multiple timezones at different times.

**Comment by thetwopct** [comment]
OK so the service I picked was just the first one I came across, so perhaps there are others. 

Even then, perhaps instead of the comprehensive PHP generated list of places of locations for Editors, we can have a 3 character timezone string text input, or a dropdown of the top 10-20 timezones, and then use `timezone_name_from_abbr("PDT")` to construct the date object. So the reverse of what you have done in `get_webinar_date_time()` to get the timezone.

But if that's too much work, I'd rather drop the feature than keep Google conversion.
